pkg/repo/migrations/v1_2.go

```diff
@@ -46,16 +46,16 @@ var V1Migration = repo.NewMigration(
 		// if no incorrect values are present they are left as is.
 		doWrite := false
 		if haveSameElements(oldSwarmPeers, cfg.Node.IPFS.SwarmAddresses) {
-			v.Set(types.NodeIPFSSwarmAddresses, []string{})
+			currentCtx.Set(types.NodeIPFSSwarmAddresses, []string{})
```

Comment:
you should keep using viper directly for the migrations as we want to work directly with config files and not with fully loaded configs with defaults and binded env variables. Also I don't think it is making things any simpler by working with config.Context